p2p/starknet/handlers.go

```diff
@@ -143,3 +160,15 @@ func (h *Handler) HandleGetTransactions(req *spec.GetTransactions) (*spec.Transa
 		Transactions: make([]*spec.Transaction, magic),
 	}, nil
 }
+
+func (h *Handler) blockByID(id *spec.BlockID) (*core.Block, error) {
+	switch {
+	case id == nil:
+		return nil, errors.New("block id is nil")
+	case id.Hash != nil:
+		hash := new(felt.Felt).SetBytes(id.Hash.Elements)
```

Comment:
Let's do this in an adaptation layer to be consistent? ie something like `p2p2core.AdaptHash`

Reply:
Done